OWML.ModHelper/ModConfig.cs

```diff
@@ -30,8 +30,16 @@ public T GetSettingsValue<T>(string key)
 
             try
             {
-                var val = value is JObject obj ? obj["value"] : value;
-                return (T)Convert.ChangeType(val, typeof(T));
+                var objectValue = value as JObject;
+                var type = typeof(T);
+                var val = (objectValue != null) ? objectValue["value"] : value;
+                if (type.IsEnum && objectValue != null && objectValue["type"].ToObject<string>() == "selector")
+                {
+                    var selected = Convert.ToString(val);
+                    var thisEnum = Enum.Parse(type, selected, true);
+                    return Enum.IsDefined(type, thisEnum) ? (T)thisEnum : default;
+                }
+                return (T)Convert.ChangeType(val, type);
```

Comment:
can this be simplified? maybe extract the selector stuff to new method? 
why is only the selector a special case here and nothing else?

Reply:
Mostly because only selector offers nice enum-like structure. Technically it would also work with any other string-based inputs, but it doesn't make much sense because of high likelihood of incorrect values

Reply:
I guess I'm partly at fault because I previously said it makes more sense to support only selector for enums. But I didn't realise it would actually simplify the code. The only difference would be the error the developer sees in case of failure. In this case we can just make the condition just `type.IsEnum` and remove the check for selector. 

Would still be good to have a separate method for converting to Enum though.

Reply:
Problem is it's impossible to do where T : enum :(